File: Cerebro Utilities/Dao/PrintingDao.cs
using Azure;
using Azure.Search.Documents;
using Azure.Search.Documents.Indexes;
using Azure.Search.Documents.Models;
using Cerebro_Utilities.Models;
using Microsoft.Extensions.Configuration;
using System;
using System.Collections.Generic;
using System.Linq;

namespace Cerebro_Utilities.Dao
{
    public interface IPrintingDao
    {
        List<PrintingEntity> GetPrintings(string cardId);
    }

    public class PrintingDao : IPrintingDao
    {
        private SearchClient _searchClient;

        private readonly IConfigurationRoot _configuration;

        public PrintingDao(IConfigurationRoot configuration)
        {
            _configuration = configuration;

            string connectionString = _configuration.GetValue<string>(Constants.CONFIG_STORAGE);

            _searchClient = new SearchIndexClient(new Uri(_configuration.GetValue<string>(Constants.INDEX_URI), UriKind.Absolute), new AzureKeyCredential(_configuration.GetValue<string>(Constants.API_KEY)))
                .GetSearchClient(PrintingEntity.INDEX_NAME);
        }

        public List<PrintingEntity> GetPrintings(string cardId)
        {
            SearchOptions options = new SearchOptions
            {
                QueryType = SearchQueryType.Full,
                SearchFields = { "ArtificialId" },
                SearchMode = SearchMode.All
            };

            Response<SearchResults<PrintingEntity>> response = _searchClient.Search<PrintingEntity>($"{cardId}*", options);
            List<PrintingEntity> entities = response.Value?.GetResults()
                .Select(x => x.Document)
                .ToList();

            List<PrintingEntity> printings = new List<PrintingEntity>();

            foreach(PrintingEntity printing in entities)
            {
                printing.Pack = PackDao._packs.Find(x => x.RowKey == printing.PackId);
                printing.Set = SetDao._sets.Find(x => x.RowKey == printing.PartitionKey);

                printings.Add(printing);
            }

            return printings;
        }
    }
}

File: dao/printingDao.js
const { SearchIndexClient, AzureKeyCredential } = require('@azure/search-documents');
const { PackDao } = require('../dao/packDao');
const { SetDao } = require('../dao/setDao');
const { PrintingEntity } = require('../models/printingEntity');

class PrintingDao {
    constructor() { }
    
    static searchClient = new SearchIndexClient(process.env.searchUri, new AzureKeyCredential(process.env.apiKey)).getSearchClient(PrintingEntity.INDEX_NAME);
    
    static async RetrieveByCard(card) {
        var searchOptions = {
            filter: `ArtificialId eq '${card.Id}'`
        };
        
        var searchResults = await this.searchClient.search('*', searchOptions);
        
        var results = [];
        
        for await (const result of searchResults.results) {
            var printing = new PrintingEntity(result.document);

            printing.Pack = PackDao.PACKS.find(pack => pack.Id === printing.PackId);
            printing.Set = SetDao.SETS.find(set => set.Name === printing.SetName);

            results.push(printing);
        }

        return results;
    };
};

module.exports = { PrintingDao };